pulsar-client-cpp/lib/LogUtils.h

```diff
@@ -35,7 +35,7 @@ namespace pulsar {
 #endif
 
 #define DECLARE_LOG_OBJECT()                                                                     \
-    static pulsar::Logger* logger() {                                                            \
+    inline pulsar::Logger* logger() {                                                            \
```

Comment:
We should not change `static` here to `inline`. Though `inline` can also avoid multiple definitions error, it will cause that a random definition is chosen in all translation units. But for each translation unit, we need a dependent definition of `logger()` because the `__FILE__` macro is different.

This change has made logs less readable, for example, see logs in https://github.com/apache/pulsar/issues/11760. All log lines' file is `ReaderTest` because it has chosen `logger()` definition from `ReaderTest.cc` for all translation units.

Reply:
@BewareMyPower ahh sorry about that!  You're quite right - I didn't notice the filenames acting screwy when I was testing this.  GCC was complaining about unused functions in several translation units with this marked "static".  I assumed it was something wrong with this definition, but I should have checked them all individually, since it's simply `DECLARE_LOG_OBJECT()` lines that are never actually used.  

If you want I can fix this in a separate PR.  I'm planning to get another change together anyway for turning on `-WExtra`.

Reply:
Since I've already added the changes in #11762 , I think we can just wait that PR to be merged.